--- a/scripts/package/mkdebian
+++ b/scripts/package/mkdebian
@@ -201,10 +201,6 @@
  This package provides userspaces headers from the Linux kernel.  These headers
  are used by the installed headers for GNU glibc and other system libraries.
 Multi-Arch: same
-EOF
-
-if is_enabled CONFIG_DEBUG_INFO; then
-cat <<EOF >> debian/control
 
 Package: $dbg_packagename
 Section: debug
@@ -218,7 +214,6 @@
 Description: Linux DTB, version $version
  This package contains device blobs from the Linux kernel, version $version
 EOF
-fi
 
 cat <<EOF > debian/rules
 #!$(command -v $MAKE) -f

